### PR TITLE
task_manager: Modify task_manager_register_task/pthread and Add TCs

### PR DESCRIPTION
**Pull request by jeongchanKim**
1. Modify registering task and pthread and fix typo
2. Add task_manager_register_task/pthread TCs

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [2219778f67236f675ecd6f7ec4453df4132203cd] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [2219778f67236f675ecd6f7ec4453df4132203cd] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [373523a83d2389f565178c97ada6f84fc1eafcc0] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [373523a83d2389f565178c97ada6f84fc1eafcc0] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [8777302ac4798ae70dec9598286835261783bbe9] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [8777302ac4798ae70dec9598286835261783bbe9] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [01aedd35b98b0edb89c68468bb868dc5bc903796] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [01aedd35b98b0edb89c68468bb868dc5bc903796] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [01aedd35b98b0edb89c68468bb868dc5bc903796] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [01aedd35b98b0edb89c68468bb868dc5bc903796] - Code Rule Check (C++) OK.

File: framework/src/task_manager/task_manager_core.c [hunk 1]
```diff
@@ -45,10 +45,12 @@ static mqd_t g_tm_recv_mqfd;
 static int builtin_cnt;
 static int handle_cnt;
 static int task_cnt;
+#ifndef CONFIG_DISABLE_PTHREAD
```
Comment (jeongarmy) [inline framework/src/task_manager/task_manager_core.c]: Does it support totally CONFIG_TASK_MANAGER_MAX_TASKS * 2 for task and thread?

**Pull request merged**